receiver/k8slogreceiver/design.md

```diff
@@ -0,0 +1,29 @@
+# Overview
+
+The k8slogreceiver is a receiver that receives logs from Kubernetes pods.
+
+```
+   K8s
+   API ─────┐
+            │
+ Docker     │
+  API ────┐ │
+          │ │
+Cri       │ │
+API ────┐ │ │
+        │ │ │
+        ▼ ▼ ▼
+    ┌──────────────┐         ┌───────────┐
+    │    Source    │────────▶│   Poller  │
+    └──────────────┘         └───────────┘
+          │                       │
+          │                       │ 
+          ▼                       │
+    ┌───────────────┐             │
+    │  Reader       │◀────────────┘
+    └───────────────┘
+```
+
+The `Source` will get information about pods from the Kubernetes API and will get the paths will log files exist from the Docker API and the CRI API (Only containerd is supported now).
```

Comment:
`dockershim` has been removed, but it is still being used for production environment for some companies and organizations. And, `cri-dockerd` is alive, which may be supported by the same implementation with `dockershim`.

Reply:
Docker api has been tested to be ok. Cri-docker will work for `daemonset-file` mode if the `graph driver` info in exposed in the API. Otherwise docker api is still needed. For cri-containerd, the snapshotter info is in a custom bytes field named "info" in the cri pod object, so we can get the root mount point.